app/services/quote_fetcher.rb

```diff
@@ -0,0 +1,18 @@
+require 'httparty'
+require 'json'
+
+class QuoteFetcher
+  include Executable
+
+  def initialize; end
+
+  def execute
+    response = JSON.parse(fetch_quotes)
+  end
+
+  private
+
+  def fetch_quotes
+    response = HTTParty.get("https://type.fit/api/quotes")
+  end
+end
```

Comment:
I like it. :grin:.

My only problem with it is we're doing a web call, not only on the partial render, but on EVERY render. We should change this to maybe a `Quote of the day` and cache the call for 24 hours.

Reply:
Check PR #163 for a caching example.

Reply:
Sorry, I meant #157.

https://github.com/mwalsh2020/lsat-cracked/blob/3f9d7b41e2a85d3dc389cd7065589b357024aa78/app/services/lawhub/authentication.rb#L21-L23

Reply:
My suggestion is:
- Make a `::quote_of_the_day` method that calls a cached `:quote_of_the_day` OR call execute.
- Make `::execute` cache a `:quote_of_the_day` that expires in 24.hours.
- I am against calling `QuoteFetcher` from the view. So I'd rather we make a helper for quotes.
- Let me know if you want to go with me on this refactoring journey on call. :joy:

Reply:
@caioertai hahaha. If you don't mind let's do a call for this refactoring journey. I think I could probably figure it out but I'd like to hear your explanation of how the caching works!